Content.Server/Implants/RadioImplantSystem.cs

```diff
@@ -0,0 +1,74 @@
+﻿using Robust.Shared.Containers;
+using Content.Shared.FixedPoint;
+using Content.Shared.Implants;
+using Content.Shared.Implants.Components;
+using Content.Shared.Tag;
+using Content.Server.Radio.Components;
+
+namespace Content.Server.Implants;
+
+public sealed class RadioImplantSystem : EntitySystem
+{
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<RadioImplantComponent, ImplantImplantedEvent>(OnImplantImplantedEvent);
+        SubscribeLocalEvent<RadioImplantComponent, EntGotRemovedFromContainerMessage>(OnRemove);
+    }
+
+    /// <summary>
+    /// If implanted with a radio implant, installs the necessary intrinsic radio components
+    /// </summary>
+    private void OnImplantImplantedEvent(EntityUid uid, RadioImplantComponent comp, ref ImplantImplantedEvent ev)
+    {
+        if (ev.Implanted == null)
+            return;
+
+        var activeRadio = EnsureComp<ActiveRadioComponent>(ev.Implanted.Value);
+
+        foreach (var channel in comp.RadioChannels)
```

Comment:
Maybe try to combine the foreach loops into one thing? I duno it might look worse if you do and or lead to weird situations... Your adding everything twice the stuff twice to `AddedChannels` which is technically fine but feels weird ya know? Totally up to you on this one 👍